src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -183,7 +189,9 @@ public IClient build() {
 			dispatcher.setMaxRequests(maxRequests);
 			dispatcher.setMaxRequestsPerHost(maxRequestsPerHost);
 
-			OkHttpClient.Builder builder = new OkHttpClient.Builder()
+			OkHttpClient.Builder builder = null;
+			if (!this.sslCertCallbackWithDefaultHostnameVerifier) {
```

Comment:
I see now you are doing `builder.hostnameVerifier(this.sslCertificateCallback)`  or not.  I would suggest:

* leaving the code as is instead of duplicating a good portion of it
* and an 'if' block after the builder assignment that evaluates setting calling `builder.hostnameVerifier(this.sslCertificateCallback)`